### PR TITLE
Fixed #15245 Added documentation & sample for xAxis.labels.allowOverlap

### LINKED ISSUE #15245: Docs: undocumented xAxis labels allowOverlap API property
Like in the title. The `xAxis.labels.allowOverlap` API property is undocumented.
https://api.highcharts.com/highcharts/xAxis.labels

#### Live demo with steps to reproduce
Uncomment `allowOverlap: true` and see the bottom labels.
https://jsfiddle.net/BlackLabel/a7ou2stn/

#### Product version
Highcharts 9.0.1

### PR DESCRIPTION
**Pull request by melanki**
https://github.com/highcharts/highcharts/issues/15245

Added API documentation for `xAxis.labels.allowOverlap`
Added sample _samples/highcharts/xaxis/labels-allowoverlap-true/_

### CODE REVIEW

**Comment by khlieng** [comment]
I think thats the first forward tick in the codebase, should probably be `'` :)

**Comment by melanki** [comment]
@khlieng All good things are three? Got the right tick this time

**Comment by khlieng** [comment]
Yes, nice!

File: samples/highcharts/xaxis/labels-allowoverlap-true/demo.details [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+ name: Highcharts Demo
+ authors:
+   - Torstein Hønsi
```
Comment (TorsteinHonsi) [inline samples/highcharts/xaxis/labels-allowoverlap-true/demo.details:4]: No

File: samples/highcharts/xaxis/labels-allowoverlap-true/demo.html [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<script src="https://code.highcharts.com/highcharts.js"></script>
+<script src="https://code.highcharts.com/highcharts-more.js"></script>
+
+<div id="container" style="height: 400px"></div>
```
Comment (TorsteinHonsi) [inline samples/highcharts/xaxis/labels-allowoverlap-true/demo.html:4]: Better add `demo.css` and do styling there.

File: ts/Core/Axis/Axis.ts [hunk 3]
```diff
@@ -1525,6 +1525,18 @@ class Axis {
              * @apioption  xAxis.labels.align
              */
 
+            /**
+             * Whether to allow the axis's labels to overlap
```
Comment (TorsteinHonsi) [inline ts/Core/Axis/Axis.ts:1529]: ... the axis labels to ...

Comment (TorsteinHonsi) [reply ts/Core/Axis/Axis.ts:1529]: And maybe elaborate: "When `false`, overlapping labels are hidden."

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Thanks for the PR! Please see the inline comments.

**Review by TorsteinHonsi — CHANGES_REQUESTED** [review]
Seems one of my comments from last time wasn't posted. The `demo.js` file doesn't lint because of inconsistent indentation and line max-len. It should have been caught in the pre-commit validation, did you commit with `--no-verify`?

**Review by TorsteinHonsi — APPROVED** [review]
Thanks!

**Pull request merged**